new-website/src/components/location/location-table.jsx

```diff
@@ -2,51 +2,93 @@ import React, { useState } from "react"
 import firebase from "gatsby-plugin-firebase"
 import MaterialTable from "material-table"
 import { LOCATION_COLLECTION } from "."
-import { useIntl } from "gatsby-plugin-intl"
+import { navigate, useIntl } from "gatsby-plugin-intl"
 import { generateI18N } from "./table-i18n"
 import { Roller } from "react-spinners-css"
+import shortid from "shortid"
 
-function choosePartnerId(entryPartnerId, userPartnerIds = []) {
-    if (entryPartnerId && userPartnerIds.includes(entryPartnerId)) {
-        return entryPartnerId
+function choosePartnerId(entryPartnerId, userPartnerIds, onError) {
+    if (entryPartnerId) {
+        if (userPartnerIds.includes(entryPartnerId)) {
+            return entryPartnerId
+        }
+        // TODO: Issue this error to sentry?
+        onError({ code: "choosePartnerId/inaccessible" })
+        throw new Error("Provided partner ID inaccessible for user")
+    } else {
+        // TODO: Depending on the context we may want users to choose and not just
+        // take the first in the sequence.
+        if (userPartnerIds.length > 0) {
+            return userPartnerIds[0]
+        }
+        // TODO: Issue this error to sentry?
+        onError({ code: "choosePartnerId/unavailable" })
+        throw new Error("No partner ID available for user")
     }
-    return userPartnerIds[0]
 }
 
-function addNewRow(userPartnerIds, onError) {
-    return async function add(data) {
-        if (!userPartnerIds || userPartnerIds.length <= 0) {
-            onError({ code: "no-permission" })
-            throw new Error("Missing Partner Id")
-        }
-        const { title, partnerId } = data
-        if (!title || title.length < 0) {
-            onError({ code: "requiredFieldMissing/title" })
-            throw new Error("missing title")
+function addNewRow(userPartnerIds, allPartnerConfigs, onError) {
+    const newPartnerLocationDoc = async () => {
+        // TODO: The partner name is a prefix in the location ID.
+        // Not sure this is ideal, but it is what we have currently.
+        const partnerId = choosePartnerId(null, userPartnerIds, onError)
+        const partnerConfig = allPartnerConfigs.find(p => p.id == partnerId)
+
+        // ShortID may collide with existing entry, retry a few times
+        for (let i=0; i<10; i++) {
+            const docId = `${partnerConfig.name}-${shortid.generate()}`
```

Comment:
Why can we not just use the random id gernerated by firestore? and let firestore take care of collisions?

Reply:
Yes, we should do that. The problem was that I encoded the pin type into the ID when hosting them in static JSON files. I did a re-import of the data and now every pin location has a type field that can be used for filtering like this:
https://github.com/ImmunHelden/ImmunHelden.de/pull/156/files#diff-1e53ef2f51bd446c876676ba83d7c888R441

Thus, I can drop the requirement for IDs to encode the type and switch to Firebase-generated ones here. IDs of imported entries still do encode their types, but from now on we can consider them to be random 😄